src/engine/strat_engine/backstore/devices.rs

```diff
@@ -843,46 +854,98 @@ mod tests {
 
         devices.push(new_info);
 
-        assert_matches!(
-            initialize_devices(devices, Uuid::new_v4(), MDADataSize::default(), None),
-            Err(_)
-        );
+        if initialize_devices(devices, Uuid::new_v4(), MDADataSize::default(), key_desc).is_ok() {
+            return Err(Box::new(StratisError::Error(
+                "Initialization should not have succeeded".to_string(),
+            )));
+        }
 
-        // Just check all paths for absence of device identifiers.
-        // Initialization of the last path was never attempted, so it should
-        // be as bare of Stratis identifiers as all the other paths that
-        // were initialized.
+        // Check all paths for absence of device identifiers or LUKS2 metadata
+        // depending on whether or not it is encrypted. Initialization of the
+        // last path was never attempted, so it should be as bare of Stratis
+        // identifiers as all the other paths that were initialized.
         for path in paths {
-            let mut f = OpenOptions::new()
-                .read(true)
-                .write(true)
-                .open(path)
-                .unwrap();
-            assert_matches!(device_identifiers(&mut f), Ok(None));
+            if key_desc.is_some() {
+                if CryptHandle::can_setup(path) {
+                    return Err(Box::new(StratisError::Error(
+                        "Device should have no LUKS2 metadata".to_string(),
+                    )));
+                }
+            } else {
+                let mut f = OpenOptions::new().read(true).write(true).open(path)?;
+                match device_identifiers(&mut f) {
+                    Ok(None) => (),
+                    _ => {
+                        return Err(Box::new(StratisError::Error(
+                            "Device should have returned nothing for device identifiers"
```

Comment:
Please put the particular device node in the error message.

Reply:
Resolved.